src/exe/win32/console.cpp

```diff
@@ -1,24 +1,46 @@
 #include "console.h"
 
+#include <iostream>
+#include <string>
+#include <system_error>
+#include <vector>
+
+#include <stdio.h>
 #include <windows.h>
 
-void attachConsole() {
-    if (GetConsoleWindow() != NULL) {
-        // Console is already attached.
-        return;
+#include "filesystem.h"
+
+static bool _haveOwnConsoleWindow = false;
+static HANDLE _consoleHelperJob = NULL;
+
+namespace xpp_win32 {
+
+static void myAllocConsole(bool hidden) {
+    // Path of Xournal++ executable
+    std::vector<wchar_t> moduleFileNameData(128, 0);
+
+    // GetModuleFileName returns number of chars excluding trailing \0 if path
+    // fits into buffer including trailing \0, else returns buffer size.
+    while (GetModuleFileNameW(NULL, moduleFileNameData.data(), moduleFileNameData.size()) ==
```

Comment:
It most likely wont ever happen, but can you guard the loop from executing infinitelly, like increasing the size only up to 2^16? It can also happen, that the function returns 0 and sets the error flag. This should be guarded too.